pkg/tools/log/logruslogger/levelchange.go

```diff
@@ -28,6 +29,21 @@ import (
 
 // SetupLevelChangeOnSignal sets the loglevel to the one specified in the map when a signal assotiated to it arrives
 func SetupLevelChangeOnSignal(ctx context.Context, signals map[os.Signal]logrus.Level) {
+	if len(signals) < 2 {
+		log.FromContext(ctx).Info("map size < 2, disabling loglevel change on signal")
+		return
+	}
+	vals := []logrus.Level{}
+	for _, v := range signals {
+		vals = append(vals, v)
+	}
+	sort.Slice(vals, func(i, j int) bool { return i < j })
+	for i := 0; i < len(vals)-1; i++ {
+		if vals[i] == vals[i+1] {
+			log.FromContext(ctx).Info("duplicate values found in map, disabling loglevel change on signal")
+			return
+		}
+	}
```

Comment:
```suggestion
I think duplicates are fine if we can do any other switch. I think we could bound this function only and only for cases when Goroutine will never do useful work.

```suggestion
var total int
for _, v := range signals{
  count[v]++
  total++
}

if total == count[logurs.GetLevel()] { 
 
 log.FromContext(ctx).WithField("logruslogger","SetupLevelChangeOnSignal").Warn("detected that log level will never changed, disabling loglevel change on signal")
  return
}
```

Thoughts?

Reply:
oh yeah, I'll edit it to get it through the ci